### PR TITLE
EG-2295 Business Networks sample CorDapp test coverage

### PR DESCRIPTION
**Pull request by tubibuto**
## Description

This PR provides full test coverage for the sample CorDapp introduced in [6546](https://github.com/corda/corda/pull/6546).

We introduce following contract tests:
- `LoanStateTest`
- `LoanContractTest`

We also build flow tests which all extend nice abstract class with helper methods under name `LoanFlowTest`:
- `AssignBICFlowTest`
- `AssignLoanIssuerRoleFlowTest`
- `IssueLoanFlowTest`
- `SettleLoanFlowTest`

## Test Plan

Ensure all existing and new tests pass.

### CODE REVIEW

**Review by johnbuckle — APPROVED** [review]

**Review by benpesterr3 — APPROVED** [review]

File: samples/business-networks-demo/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Attachment Demo
```
Comment (bpaunescu) [inline samples/business-networks-demo/README.md:1]: Why does this have to do with Attachments?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalTestUtils.kt [hunk 2]
```diff
@@ -262,12 +262,12 @@ class ListenProcessDeathException(hostAndPort: NetworkHostAndPort, listenProcess
         CordaException("The process that was expected to listen on $hostAndPort has died with status: ${listenProcess.exitValue()}")
 
 class NodeListenProcessDeathException(hostAndPort: NetworkHostAndPort, listenProcess: Process, val causeFromStdError: String) :
-    CordaException(
-        """
+        CordaException(
```
Comment (bpaunescu) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalTestUtils.kt]: The BN work will be moved outside of corda, so I don't think it should require any changes done to Corda internals (even formatting ones :P )

**Review by bpaunescu — COMMENTED** [review]
A couple of questions only.

File: samples/business-networks-demo/README.md (hunk 1 above)
Comment (tubibuto) [reply samples/business-networks-demo/README.md:1]: Oh this happens when you submit PRs late Friday evening. I have to remove this README contents and will submit separate PR with proper README

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalTestUtils.kt (hunk 2 above)
Comment (tubibuto) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalTestUtils.kt]: Similar answer to the one above.

**Pull request merged**